packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -17,19 +17,30 @@ describe('<DataGrid /> - Filter', () => {
         id: 0,
         brand: 'Nike',
         isPublished: false,
+        country: 'United States',
       },
       {
         id: 1,
         brand: 'Adidas',
         isPublished: true,
+        country: 'Germany',
       },
       {
         id: 2,
         brand: 'Puma',
         isPublished: true,
+        country: 'Germany',
+      },
+    ],
+    columns: [
+      { field: 'brand' },
+      { field: 'isPublished', type: 'boolean' },
+      {
+        field: 'country',
+        type: 'select',
+        valueOptions: ['United States', 'Germany', 'France'],
```

Comment:
what if you want to render a template in the select? Ie flag and country?
Also I would have expected the grid to just generate the options by getting all the different values of a column without needing to provide those options.

Reply:
I thought about that but what if there is a value that no column has, then it is a problem of not having a way go get it.

Regarding the template - based on that discussion https://github.com/mui-org/material-ui-x/pull/1956#discussion_r658350955 we can start simple and extend it. For custom templates, the column type can be extended. I can have the same argument for the date picker and for any of the other column types.